src/Core/Foundation/IoC/Container.php

```diff
@@ -104,9 +104,8 @@ private function makeInstanceFromClassName($className, array $alreadySeen)
 
         if ($classConstructor) {
             foreach ($classConstructor->getParameters() as $param) {
-                $paramClass = $param->getClass();
-                if ($paramClass) {
-                    $args[] = $this->doMake($param->getClass()->getName(), $alreadySeen);
+                if ($param->getType() instanceof \ReflectionNamedType && !$param->getType()->isBuiltin()) {
```

Comment:
Could put the class `ReflectionNamedType` in `use` section ?

Reply:
Please do so and squash & merge.

Reply:
Please do it and I will take time to review your PR.

Reply:
Done, please take your time for my other PRs as well.